src/networkd.c

```diff
@@ -475,6 +476,15 @@ write_network_file(net_definition* def, const char* rootdir, const char* path)
         dhcp_overrides combined_dhcp_overrides;
         combine_dhcp_overrides(def, &combined_dhcp_overrides);
 
+        /* Only set MTU from DHCP if use-mtu dhcp-override is not false. */
+        if (!combined_dhcp_overrides.use_mtu) {
+            /* isc-dhcp dhclient compatible UseMTU, networkd default is to
+             * not accept MTU, which breaks clouds */
+            g_string_append_printf(s, "\n[DHCP]\nUseMTU=false\n");
+        } else {
+            g_string_append_printf(s, "\n[DHCP]\nUseMTU=true\n");
+        }
+
```

Comment:
Please keep [DHCP] title added earlier, and simply write out UseMTU=whatever. There are examples of how this has been done elsewhere. Otherwise some other values will be included in another section than the "DHCP" section.

Please also adjust the tests to make sure they still pass (will involve moving the UseMTU= line in expected outputs).

Reply:
Ah, good catch. Will fix this.

Reply:
Does the new diff look okday?